routes/recommend.py

```diff
@@ -1,54 +1,75 @@
 from fastapi import APIRouter, UploadFile, File, Form, HTTPException
 from fastapi.responses import JSONResponse
-from typing import Optional
-from PIL import Image
-import io, base64
+from utils.image_upload import upload_to_catbox
+import tempfile
+import os
+import requests
 
 router = APIRouter()
 
+SERPAPI_API_KEY = "1a0a6ff203243d30332842bd64f7a3ee9f01dce373cb7e1af1b40e3319f6f7e8"
+
 @router.post("/recommend")
 async def recommend_from_image(
     image: UploadFile = File(...),
     prompt: str = Form(...)
 ):
     try:
-        contents = await image.read()
-        input_image = Image.open(io.BytesIO(contents)).convert("RGB")
-
-        # 더미 데이터 반환 
-        dummy_results = [
-            {
-                "image": "data:image/png;base64,base64encodedimagedata1",
-                "price": "120,000원",
-                "address": "https://map.naver.com/v5/entry/place/123456"
-            },
-            {
-                "image": "data:image/png;base64,base64encodedimagedata2",
-                "price": "85,000원",
-                "address": "https://map.naver.com/v5/entry/place/234567"
-            },
-            {
-                "image": "data:image/png;base64,base64encodedimagedata3",
-                "price": None,
-                "address": "https://map.naver.com/v5/entry/place/345678"
-            },
-            {
-                "image": "data:image/png;base64,base64encodedimagedata4",
-                "price": "45,000원",
-                "address": "https://map.naver.com/v5/entry/place/456789"
-            },
-            {
-                "image": "data:image/png;base64,base64encodedimagedata5",
-                "price": None,
-                "address": "https://map.naver.com/v5/entry/place/567890"
+        suffix = os.path.splitext(image.filename)[1]
+        with tempfile.NamedTemporaryFile(delete=False, suffix=suffix) as tmp:
+            tmp.write(await image.read())
+            tmp_path = tmp.name
+
+        image_url = upload_to_catbox(tmp_path)
```

Comment:
현재는 serpAPI를 쓰고 있어 upload_to_catbox라는 함수를 호출하여 image_url을 받아오지만, 후에 구현 API가 바뀌게 되면 해당 로직은 전부 변경되어야 하는 리스크가 있습니다.

따라서 image_url을 구하는 로직을 get_image_url로 캡슐화하고, 구현 API변경시 내부 함수만 변경할수 있도록 구조를 개선하는 것이 어떠세요?

```suggestion
        image_url = upload_to_catbox(tmp_path)
        image_url = get_image_url(tmp_path)
        
        def get_image_url(temp_path):
            ### API가 바뀌면 여기만 수정하면됨
            return upload_to_catbox(tmp_path)
            
```